packages/calling-stateful-client/src/CallContext.ts

```diff
@@ -574,7 +574,7 @@ export class CallContext {
          * camera permissions are granted. So selectedCamera will have this value before the actual cameras are obtained. Therefore we should reset
          * selectedCamera to the first camera when there are cameras AND when current selectedCamera does not exist in the new array of cameras **/
         if (cameras.length > 0 && !cameras.some((camera) => camera.id === draft.deviceManager.selectedCamera?.id)) {
-          this.setDeviceManagerSelectedCamera(cameras[0]);
+          draft.deviceManager.selectedCamera = cameras[0];
```

Comment:
Oops. Good catch.

Makes sense that if `immer::produce()` is not reentrant. `this._state` is overwritten with the draft here at the end of the block, so any updates by `setDeviceManagerSelectedCamera` will be lost.